pilot/pkg/bootstrap/config_compare.go

```diff
@@ -35,27 +36,11 @@ func needsPush(prev config.Config, curr config.Config) bool {
 	if !strings.HasSuffix(prev.GroupVersionKind.Group, "istio.io") {
 		return true
 	}
-	// If current/previous metadata has "*istio.io" label/annotation, just push
```

Comment:
This was not only for "AlwaysPush". There are annotations that impact XDS generation. Below we only compare spec, so if a config changes just the annotation nothing happens.

Things like istio.io/dry-run, etc

Reply:
why should make dryrun push?

Reply:
Any change to inputs that produces a change in outputs should result in a push so things are recomputed. Otherwise the update would be ignored and generate invalid information.

Its not special about "dry-run", just anything where annotation change leads to XDS changes. I recall dry-run was one motivation use

Reply:
Can you be more explicit, i donot think of a case we should annotate a config to make istiod push.

Originally this function is to mitigate pushed when labels change to a DR. Simutabeously introduce alwaysPush label, but not sure i know why "*istio.io" should be checked if we checked alwaysPush labels explicitly

Reply:
We check istio.io only because we know other random annotations are not used in xds generation. It's a hack to improve performance and ignore irrelevant annotations

Reply:
..... any example?

Reply:
https://github.com/istio/istio/blob/c5463a0b9e381842049a2f2081f361affc2dd0a2/pilot/pkg/security/authz/builder/builder.go#L162

Reply:
I just learned another annotation, hha.  This makes me hard to do any simplify. 

BTW, if we need to keep the feature or this is a useful feature, i think so.  we should move it under spec.